financial-connections/src/main/java/com/stripe/android/financialconnections/FinancialConnectionsSheetRedirectActivity.kt

```diff
@@ -30,27 +31,49 @@ class FinancialConnectionsSheetRedirectActivity : AppCompatActivity() {
     /**
      * @return Intent, or null if deeplink cannot be mapped to an Intent (in case of unknown deep links).
      */
-    private fun Uri.toIntent(): Intent? =
-        when {
-            isFinancialConnectionsScheme().not() -> null
-            // auth-redirect hosts: redirections from Abstract Auth in web back to native SDK
-            host == "auth-redirect" ->
-                Intent(
-                    this@FinancialConnectionsSheetRedirectActivity,
-                    FinancialConnectionsSheetNativeActivity::class.java
-                )
-            // link-accounts hosts:
-            // - redirections from embedded web AuthFlow back to SDK (app2app return and AuthFlow end)
-            // native-redirect hosts:
-            // - redirections from embedded web AuthFlow back SDK (app2app start)
-            host == "link-accounts" || host == "native-redirect" -> Intent(
-                this@FinancialConnectionsSheetRedirectActivity,
-                FinancialConnectionsSheetActivity::class.java
-            )
-            else -> null
+    private fun Uri.toIntent(): Intent? = when {
+        isFinancialConnectionsScheme().not() -> null
+        // auth-redirect hosts:
+        // redirections from Abstract Auth (shim) in web back to native SDK
+        host == HOST_AUTH_REDIRECT -> FinancialConnectionsSheetNativeActivity::class.java
+        // native-link-accounts hosts:
+        // redirections from app2app back to SDK (while in native flow)
+        host == HOST_NATIVE_LINK_ACCOUNTS -> FinancialConnectionsSheetNativeActivity::class.java
+        // link-accounts hosts:
+        // - redirections from embedded web AuthFlow back to SDK (/success, /cancel, /fail)
+        // - redirections from app2app back to SDK (while in web AuthFlow)
+        host == HOST_LINK_ACCOUNTS -> FinancialConnectionsSheetActivity::class.java
+        // native-redirect hosts:
+        // redirections from embedded web AuthFlow back SDK (app2app start)
+        host == HOST_NATIVE_REDIRECT -> FinancialConnectionsSheetActivity::class.java
+
+        else -> null
+    }?.let { destinationActivity ->
+        Intent(
+            this@FinancialConnectionsSheetRedirectActivity,
+            destinationActivity
+        )
+    }
```

Comment:
Same logic, but

- Extracting some strings to constants
- Extracting the duplicated `Intent` creation